SCF-Session5-Question1/src/meta/utility/Maths.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This class contains recursive methods to calculate LCM & HCF
+ * @author ABHISHEK BISHNOI
+ * @version 1.0
+ * since 18-July-2019
+ */
+package meta.utility;
+
+public class Maths {
+
+	 /*
+	 * this function calculates hcf of 2 values using Euclid Algorithm
+	 * To know more About Euclid Algorithm follow this 
+	 * https://en.wikipedia.org/wiki/Euclidean_algorithm
+	 * @param int this is the first value for hcf opertaion
+	 * @param int int this is the second value for hcf opertaion
+	 * return int returns hcf 
+	 * Assumption All Values Are Positive
+	 */
+	static int flag =-1;
+	public static int hcf( int firstValue, int secondValue ){ // hcf(3,5)
```

Comment:
Few points:
1. What is the use of flag ?
2. Why checking (firstValue==0 || secondValue == 0) condition in the method ? We can check this before calling the hcf method ?
3. Can't we use a single return statement in this method ?
4. Can't we structure this function in more compact and cleaner way ?